src/main/scala/db/migration/V25__RemoveDomainFromH5PUrl.scala

```diff
@@ -0,0 +1,114 @@
+/*
+ * Part of NDLA draft-api.
+ * Copyright (C) 2020 NDLA
+ *
+ * See LICENSE
+ */
+
+package db.migration
+
+import org.flywaydb.core.api.migration.{BaseJavaMigration, Context}
+import org.json4s
+import org.json4s.DefaultFormats
+import org.json4s.JsonAST.{JArray, JString}
+import org.json4s.native.JsonMethods.{compact, parse, render}
+import org.postgresql.util.PGobject
+import scalikejdbc.{DB, DBSession, _}
+
+class V25__RemoveDomainFromH5PUrl extends BaseJavaMigration {
+  implicit val formats: DefaultFormats.type = org.json4s.DefaultFormats
+
+  override def migrate(context: Context) = {
+    val db = DB(context.getConnection)
+    db.autoClose(false)
+
+    db.withinTx { implicit session =>
+      migrateArticles
+    }
+  }
+
+  def migrateArticles(implicit session: DBSession): Unit = {
+    val count = countAllArticles.get
+    var numPagesLeft = (count / 1000) + 1
+    var offset = 0L
+
+    while (numPagesLeft > 0) {
+      allArticles(offset * 1000).map {
+        case (id, document) => updateArticle(convertArticleUpdate(document), id)
+      }
+      numPagesLeft -= 1
+      offset += 1
+    }
+  }
+
+  def countAllArticles(implicit session: DBSession): Option[Long] = {
+    sql"select count(*) from articledata where document is not NULL"
+      .map(rs => rs.long("count"))
+      .single()
+      .apply()
+  }
+
+  def allArticles(offset: Long)(implicit session: DBSession): Seq[(Long, String)] = {
+    sql"select id, document, article_id from articledata where document is not null order by id limit 1000 offset $offset"
+      .map(rs => {
+        (rs.long("id"), rs.string("document"))
+      })
+      .list
+      .apply()
+  }
+
+  def updateArticle(document: String, id: Long)(implicit session: DBSession): Int = {
+    val dataObject = new PGobject()
+    dataObject.setType("jsonb")
+    dataObject.setValue(document)
+
+    sql"update articledata set document = $dataObject where id = $id"
+      .update()
+      .apply
+  }
+
+  def updateH5PDomains(html: String): String = {
+    val oldDomain = "https?:\\/\\/h5p.{0,8}.ndla.no"
+    val newDomain = ""
+
+    val updatedHtml = html.replaceAll(oldDomain, newDomain)
+    updateH5PResourceType(updatedHtml)
+  }
+
+  def updateH5PResourceType(html: String): String = {
+    val oldResourceType = "external"
+    val newResourceType = "h5p"
+
+    html.replaceAll(oldResourceType, newResourceType)
```

Comment:
Denne vil gjøre _alle_ external embeds til h5p? Det hadde vært ganske uheldig da det finnes mange external embeds som ikke er h5p :)
Den vil forsåvidt også gjøre om `external` til `h5p` alle steder det skrives i teksten også. Det kan jo være litt uheldig. 
Det [finnes](https://ndla.no/subjects/subject:39/topic:1:189089/resource:1:121203) artikler hvor det hadde vært kjipt :)

Reply:
Litt for fort i svingene her, `convertArticleUpdate` parser dokumentet som den får som parameter og inni der så begynner vi å loope på elementene i artikkelen. For TopicArticle ser vi etter visualElement som har en resource inni seg og der er da og bare da vi begynner å bruke `updateH5PDomains` og `updateH5PResourceType`. De påvirker ikke resten av artikkelen.

Reply:
Veldig sant. Det er kun h5p-embeds som må konverteres.